motionplan/motionPlanner.go

```diff
@@ -362,7 +420,17 @@ IK:
 		}
 	}
 	if len(solutions) == 0 {
-		return nil, errIKSolve
+		// We have failed to produce a usable IK solution. Let the user know if zero IK solutions were produced, or if non-zero solutions
+		// were produced, which constraints were failed
+		if constraintFailCnt == 0 {
+			return nil, errIKSolve
+		}
+		errIKConstraint := "all IK solutions failed constraints. Failures: "
```

Comment:
I think we should make this its own function and keep all error message related code in `motionplan/errors.go`.  It could take in the map you've built and return the full error message

Reply:
Yah, good call. Moving it over.